packages/controllers/src/snaps/SnapController.ts

```diff
@@ -645,7 +645,7 @@ export class SnapController extends BaseController<
         this._maxIdleTime &&
         timeSince(runtime.lastRequest) > this._maxIdleTime
       ) {
-        this.stopSnap(snapId, SnapStatusEvent.stop);
+        await this.stopSnap(snapId, SnapStatusEvent.stop);
```

Comment:
Why were `await`s added for various `stopSnap` calls?

Reply:
`stopSnap` was marked as `async` before but didn't have async logic. Before, it was safe to run it without awaiting because of this. But now it does have actual async logic and therefore we need to `await` it. We expect the snap to be stopped once we continue execution of the functions in question.

Reply:
We should use `Promise.allSettled` here instead, and we should also consider whether we want to do that in other places that use `Promise.all`.

Reply:
`Promise.all` will execute all asynchronous function calls, and either resolve with an array of all results or reject with the error of the first promise to reject. That being the case, it will swallow any rejections subsequent to the first one. If we want to make sure we capture every rejection, we would have to use `Promise.allSettled` and process the resulting array, which contains values indicating whether each promise was resolved or rejected.

Ref:
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled